haystack/components/tools/tool_invoker.py

```diff
@@ -156,12 +159,13 @@ def dummy_weather_function(city: str):
     print(result)
     """
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-positional-arguments
         self,
         tools: Union[List[Tool], Toolset],
         raise_on_failure: bool = True,
         convert_result_to_json_string: bool = False,
         streaming_callback: Optional[StreamingCallbackT] = None,
+        async_executor: Optional[ThreadPoolExecutor] = None,
```

Comment:
Let's remove the disable and instead enforce keyword only args for the new option you are adding so 
```suggestion
    def __init__(
        self,
        tools: Union[List[Tool], Toolset],
        raise_on_failure: bool = True,
        convert_result_to_json_string: bool = False,
        streaming_callback: Optional[StreamingCallbackT] = None,
        *,
        async_executor: Optional[ThreadPoolExecutor] = None,
```